src/components/course/data/utils.jsx

```diff
@@ -452,38 +452,49 @@ export const getCouponCodesDisabledEnrollmentReasonType = ({
   return undefined;
 };
 
+/**
+ * Determines whether the learner's subscription license is applicable to the course being
+ * viewed, based on the enterprise catalogs associated with the learner's subscription license.
+ * @param {Object} args
+ * @param {array} args.catalogsWithCourse List of catalogs that will be checked against the license.
+ * @param {Object} args.subscriptionLicense Learner's subscription license.
+ * @returns {boolean} True if the learner's subscription license is applicable to the course being viewed.
+ */
+function determineLicenseApplicableToCourse({
+  catalogsWithCourse,
+  subscriptionLicense,
+}) {
+  if (!subscriptionLicense) {
+    return false;
+  }
+  return catalogsWithCourse.includes(
+    subscriptionLicense.subscriptionPlan.enterpriseCatalogUuid,
+  );
+}
+
 export const getSubscriptionDisabledEnrollmentReasonType = ({
   customerAgreement,
   catalogsWithCourse,
   subscriptionLicense,
   hasEnterpriseAdminUsers,
 }) => {
-  if (!subscriptionLicense) { return undefined; }
-  const hasExpiredSubscriptionLicense = !subscriptionLicense.subscriptionPlan.isCurrent;
-  if (hasExpiredSubscriptionLicense) {
-    return parseReasonTypeBasedOnEnterpriseAdmins({
-      hasEnterpriseAdminUsers,
-      reasonTypes: {
-        hasAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_EXPIRED,
-        hasNoAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_EXPIRED_NO_ADMINS,
-      },
-    });
-  }
```

Comment:
non-blocking question: Why move this check to the end?  Didn't the original ticket suggest that re-prioritizing the checks in `getMissingApplicableSubsidyReason` would be sufficient?  Does this change have an impact?